Remove line limits

ajstorm opened the pull request:
This PR pulls in 0b1645d91e851e735d3e23330303ce81f70adbe3 and 7649d4548cb53a614db133b2a8ac1f31859dda8c from the upstream repo, which collectively allow for configurable line limits.

Line limits are causing issues on in cockroachdb/cockroach when printing
out long constraints fields for zone configurations. With limits in
place, long zone configuration constraints fields will have one or more
`\n`'s added to them. To avoid this, and have the constraints field
appear as one contiguous line (regardless of its length), we remove the
line limits here.

ajstorm committed:
Remove line limits

This commit pulls in 0b1645d91e851e735d3e23330303ce81f70adbe3 from the
upstream repo, which removes line limits.

Line limits are causing issues on in cockroachdb/cockroach when printing
out long constraints fields for zone configurations. With limits in
place, long zone configuration constraints fields will have one or more
`\n`'s added to them. To avoid this, and have the constraints field
appear as one contiguous line (regardless of its length), we remove the
line limits here.

rafiss reviewed (changes requested):
it looks like the upstream added one more commit that makes it configurable: https://github.com/go-yaml/yaml/commit/7649d4548cb53a614db133b2a8ac1f31859dda8c

can we add this `FutureLineWrap` option in this PR as well, so that CRDB can go back to the old behavior more easily if necessary? (up to you if you want to make line wrapping enabled or disabled by default)

ajstorm committed:
Pull in commit which reverts default line length change

This commit pulls in 7649d45 which reverts the default line length change and
instead makes the line length configurable. See parent commit for more details.

ajstorm commented:
Good call.  I noticed that commit too, but thought it might not be worth bringing it along for the ride.  You've convinced me otherwise, as it'll be best if we keep the line lengths as-is by default.

rafiss reviewed (approved):
lgtm!
